### PR TITLE
Project restructure + cross-building

### PR DESCRIPTION
**Pull request by wajda**
As agreed Commons will be a single-module project containing classes logically grouped by packages, with all transitive dependencies marked as "optional".

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,20 +76,166 @@
...
+                        <goals>
+                            <goal>testCompile</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>com.github.wajda</groupId>
+                <artifactId>scala-cross-build-maven-plugin</artifactId>
+                <version>0.2.0</version>
```
Comment (yruslan) [inline pom.xml]: 😮  Wow. You did release that plugin to Maven Central, didn't you?

Comment (wajda) [reply pom.xml]: Upon request :)
https://github.com/Stratio/scala-cross-build-maven-plugin/issues/46

File: build-all.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  find $BASE_DIR/target/* -type d -exec rm -rf {} \;
+  mvn scala-cross-build:change-version -Pscala-$SCALA_VER
+  mvn install
+}
+
+# ------------------------------------------------
+
+mvn clean
+
+cross_build 2.11
```
Comment (yruslan) [inline build-all.sh]: What if there are compilation errors or test failures during a build for a particular Scala version? 

Should it check the exit code and don't start the next build if the previous one failed?

Comment (wajda) [reply build-all.sh]: no, no. This shell script is only there for dev convenience.  If I want to quickly run all builds at once locally I run this script. It's not intended for anything else.
On CI a separate build is created per Scala version

File: pom.xml [hunk 3]
```diff
@@ -13,20 +13,49 @@
...
+    <name>ABSA Commons</name>
+    <description>Selection of useful reusable components</description>
+    <inceptionYear>2019</inceptionYear>
+    <url>https://absaoss.github.io/commons/</url>
+
     <groupId>za.co.absa.commons</groupId>
-    <artifactId>commons</artifactId>
-    <packaging>pom</packaging>
+    <artifactId>commons_2.11</artifactId>
     <version>1.0.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (wajda) [reply pom.xml]: Good point. Let's start with 0.0.1 and then switch to 0.1.0 when decent amount of the stuff is added.

**Pull request merged**